### PR TITLE
PS-5665 Documentation missing for binlog_space_limit in 8.0

### PR DESCRIPTION
**Pull request by patrickbirch**
Added missing page, updated index and checked ps-versions-comparison doc.

### CODE REVIEW

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
It is max_binlog_size that needs to be documented
The max_binlog_files variable has been removed from 8.0 and only should be mentioned, if ever, historically as a 5.7 feature that's no longer present.

Please also read the internal Confluence page on commit message formatting

**Comment by patrickbirch** [comment]
deleted source/flexibility/max_binlog_files.rst
renamed, rewrote, and updated source/flexibility/maximum_binlog_size.rst
updated source/index.rst replacing max_binlog_files with maximum_binlog_size.rst
updated source/ps-variables with variables
updated the link for source/ps-versions-comparison

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (dutow) [inline doc/source/flexibility/maximum_binlog_size.rst]: Why are there 5.7 versions mentioned in an 8.0 documentation? Especially about a removed variable.

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (dutow) [inline doc/source/flexibility/maximum_binlog_size.rst]: This is an upstream variable? https://dev.mysql.com/doc/refman/8.0/en/replication-options-binary-log.html#sysvar_max_binlog_size

File: doc/source/ps-variables.rst
Comment (dutow) [inline doc/source/ps-variables.rst]: This is an upstream variable.

File: doc/source/ps-versions-comparison.rst
Comment (dutow) [inline doc/source/ps-versions-comparison.rst]: We no longer restring the number of files - only the size of the files.

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (dutow) [inline doc/source/flexibility/maximum_binlog_size.rst]: * Not only a single transaction: as the documentation later describes, the last binlog file will never be pruned.
* Please use consistent grammar, e.g. With variable set..., you set, ...

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (dutow) [inline doc/source/flexibility/maximum_binlog_size.rst]: "default value and the maximum value and the <variable>" - there are definitely some extra words here.

I also don't understand the example - why three? Why does it have to be one transaction? And it doesn't mention the current binlog size, only that the maximum is 10g - so if currently the sum of binlog size is 3 or 4g, this variable will do nothing.

File: doc/source/ps-versions-comparison.rst
Comment (dutow) [inline doc/source/ps-versions-comparison.rst:51]: Why changed?

**Review by dutow — CHANGES_REQUESTED** [review]

**Comment by dutow** [comment]
Please also change your commit message. The first line should contain the jira ticket id, otherwise jira won't pick it up. E.g. `PS-5665: ...` or `PS-5665 ...`

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (patrickbirch) [reply doc/source/flexibility/maximum_binlog_size.rst]: removed.

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (patrickbirch) [reply doc/source/flexibility/maximum_binlog_size.rst]: Added a link to the MySQL documentation.

File: doc/source/ps-variables.rst
Comment (patrickbirch) [reply doc/source/ps-variables.rst]: removed from the list of ps-variables.

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (patrickbirch) [reply doc/source/flexibility/maximum_binlog_size.rst]: rewritten.

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (patrickbirch) [reply doc/source/flexibility/maximum_binlog_size.rst]: Removed most of the example text.

File: doc/source/ps-versions-comparison.rst
Comment (patrickbirch) [reply doc/source/ps-versions-comparison.rst:51]: Changed the "number" to "size" since you said that with 8.0 "we are no longer restricting the number"..."only the size"

**Comment by patrickbirch** [comment]
Updated commit message and PR heading.

File: doc/source/ps-versions-comparison.rst
Comment (patrickbirch) [reply doc/source/ps-versions-comparison.rst]: changed to restricting the size of binlog files

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (dutow) [inline doc/source/flexibility/maximum_binlog_size.rst:61]: I don't see an upper limit in the code, it's ULONG_MAX

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (dutow) [inline doc/source/flexibility/maximum_binlog_size.rst:44]: 4096-1GB isn't realy clear (as 4096 is larger than 1). It would be better to keep this the same as upstream, using 1073741824 instead of 1GB.

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (patrickbirch) [reply doc/source/flexibility/maximum_binlog_size.rst:61]: fixed.

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (patrickbirch) [reply doc/source/flexibility/maximum_binlog_size.rst:44]: Fixed.

**Comment by patrickbirch** [comment]
waiting for either approval or comments.

File: doc/source/flexibility/maximum_binlog_size.rst
Comment (dutow) [reply doc/source/flexibility/maximum_binlog_size.rst:61]: This wasn't fixed, range still shows "0-102400" 
There is no limit other than the vartype.

Please see how upstream documents these, like binlog_cache_size: https://dev.mysql.com/doc/refman/5.6/en/replication-options-binary-log.html#sysvar_binlog_cache_size

**Review by dutow — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
